core/src/smartcontracts/isi/permissions.rs

```diff
@@ -422,74 +910,206 @@ impl<W: WorldTrait, O: NeedsPermission> IsAllowed<W, O> for AnyShouldSucceed<W,
         Err(format!(
             "None of the instructions succeeded in Any permission check block with name: {}",
             self.name
-        ))
+        )
+        .into())
     }
 }
 
-impl<W: WorldTrait, O: NeedsPermission + 'static> From<AnyShouldSucceed<W, O>>
-    for IsAllowedBoxed<W, O>
-{
-    fn from(validator: AnyShouldSucceed<W, O>) -> Self {
-        Box::new(validator)
+impl IsAllowed<World, QueryBox> for AnyShouldSucceed {
```

Comment:
This file should probably be split into modules (which could be their own files).